katello/hooks/pre/30-upgrade.rb

```diff
@@ -37,6 +37,28 @@ def migrate_foreman
   Kafo::Helpers.execute('foreman-rake db:migrate')
 end
 
+def postgresql_10_upgrade
+  #Ensure there is at least 1x /var/lib/pgsql free disk space available on the main filesystem
+  POSTGRES_DIR = '/var/lib/pgsql'
+
+  begin
+    available_space = `df -k --total --exclude-type=tmpfs --output=avail`.split("\n")[-1].to_i
+    postgres_size = File.directory?(POSTGRES_DIR) ? `du -k -s #{POSTGRES_DIR}`.split[0].to_i : 0
+    if available_space < postgres_size
+      fail_and_exit 'The postgres upgrade requires at least #{postgres_size} KB of storage.'
+    end
+  rescue StandardError
+    fail_and_exit 'Failed to verify available disk space'
+  end
+
+  Kafo::Helpers.execute('yum -y install rh-postgresql10-postgresql-server')
+  Kafo::Helpers.execute('foreman-maintain service stop')
+  Kafo::Helpers.execute('scl enable rh-postgresql10 -- postgresql-setup --upgrade')
+  Kafo::Helpers.execute('yum -y remove postgresql postgresql-server')
+  Kafo::Helpers.execute('rm -f /etc/systemd/system/postgresql.service')
+  Kafo::Helpers.execute('yum -y install rh-postgresql10-syspaths')
```

Comment:
Since this is a pre hook, the installer runs after it. That ensures all `-syspaths` are installed so is this really needed? Does `rh-postgresql10-syspath` even exist as a package?

Reply:
We want syspaths for the services and the client bits so I think using the overall package is ok.

```
rh-postgresql10-syspaths.x86_64 : System-wide wrappers for the rh-postgresql10
```